packages/phishing-controller/src/PhishingDetector.test.ts

```diff
@@ -0,0 +1,979 @@
+      await withPhishingDetector(
+        [
+          {
+            allowlist: ['allowed-first-fuzzy-second.com'],
+            blocklist: [],
+            fuzzylist: [],
+            name: 'first',
+            tolerance: 2,
+            version: 1,
+          },
+          {
+            allowlist: [],
+            blocklist: [],
+            fuzzylist: ['allowed-first-fuzzy-second.com'],
+            name: 'second',
+            tolerance: 2,
+            version: 1,
+          },
+        ],
+        async ({ detector }) => {
+          const { result, type, name } = detector.check(
+            'allowed-first-fuzzy-second.com',
+          );
+
+          expect(result).toBe(false);
+          expect(type).toBe('allowlist');
+          expect(name).toBe('first');
+        },
+      );
+    });
+
+    it('blocks a blocklisted domain when it ends with a dot', async () => {
+      await withPhishingDetector(
+        [
+          {
+            allowlist: [],
+            blocklist: ['blocked.com'],
+            fuzzylist: [],
+            name: 'first',
+            tolerance: 2,
+            version: 1,
+          },
+        ],
+        async ({ detector }) => {
+          const { result, type } = detector.check('blocked.com.');
+
+          expect(result).toBe(true);
+          expect(type).toBe('blocklist');
+        },
+      );
+    });
+
+    describe('with legacy config', () => {
+      it('changes the type to whitelist when the result is allowlist', async () => {
+        await withPhishingDetector(
+          {
+            blacklist: [],
+            fuzzylist: [],
+            tolerance: 2,
+            whitelist: ['allowed.com'],
+          },
+          async ({ detector }) => {
+            const { type, result } = detector.check('allowed.com');
+
+            expect(type).toBe('whitelist');
+            expect(result).toBe(false);
+          },
+        );
+      });
+
+      it('changes the type to blacklist when the result is blocklist', async () => {
+        await withPhishingDetector(
+          {
+            blacklist: ['blocked.com'],
+            fuzzylist: [],
+            tolerance: 2,
+            whitelist: [],
+          },
+          async ({ detector }) => {
+            const { type, result } = detector.check('blocked.com');
+
+            expect(type).toBe('blacklist');
+            expect(result).toBe(true);
+          },
+        );
+      });
+    });
+  });
+});
+
+type WithPhishingDetectorCallback<ReturnValue> = ({
+  detector,
+}: {
+  detector: PhishingDetector;
+}) => Promise<ReturnValue> | ReturnValue;
+
+type WithPhishingDetectorArgs<ReturnValue> = [
+  PhishingDetectorOptions,
+  WithPhishingDetectorCallback<ReturnValue>,
+];
+
+/**
+ * Build a phishing detector and run a callback with it.
+ *
+ * @param args - The phishing detector options and callback.
+ * @returns The return value of the callback.
+ */
+async function withPhishingDetector<ReturnValue>(
```

Comment:
I don't think we need a clean-up after each test run, but I added this because ideally if all other classes under test use this pattern then it should be more easily recognizable and readable by a developer reading it for the first time.

But I don't have a strong opinion, we can also keep things "simple" and instantiate manually the detector in each test case